news/11325.feature.rst

```diff
@@ -0,0 +1 @@
+Support a per-requirement ``--config-settings`` option.
```

Comment:
I think this should mention it’s relevant for requirements files, otherwise it’s a bit confusing (You can’t supply different config settings in CLI e.g. `pip install foo bar`?)

Reply:
Updated